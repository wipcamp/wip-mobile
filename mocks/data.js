export const newProblem = {
    topic: "Test new problem",
    problem_type_id: 1,
    description: "Test new problem",
    report_id: 1
}

export const problem = [
    {
        id: 1,
        topic: "Test Problem 1",
        description: "Test Description 1",
        problem_type_id: 1,
        report_id: 1,
        is_solve: false,
        not_solve: false,
        created_at: "2018-02-02 17:50:01",
        updated_at: "2018-02-02 17:50:01"
    },
    {
        id: 2,
        topic: "Test Problem 2",
        description: "Test Description 2",
        problem_type_id: 2,
        report_id: 2,
        is_solve: false,
        not_solve: false,
        created_at: "2018-02-02 17:50:02",
        updated_at: "2018-02-02 17:50:02"
    }
]

export const categoryProblem = [
    {
        value: 1,
        label: "Test 1"
    },
    {
        value: 2,
        label: "Test 2"
    }
]

export const filter = 0

export const filterView = [
    {
        value: 0,
        label: "ทั้งหมด"
    },
    {
        label: 1,
        value: "Test 1"
    },
    {
        label: 2,
        value: "Test 2"
    }
]