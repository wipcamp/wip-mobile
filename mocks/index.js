const mockComponent = require.requireActual('react-native/jest/mockComponent')

module.exports = {
    initAll: () => {
      jest.mock('ScrollView', () => {
        const RealScrollView = require.requireActual('ScrollView')
        const ScrollView = mockComponent('ScrollView')
        ScrollView.prototype.render = RealScrollView.prototype.render
        return ScrollView
      })

      jest.mock('Dimensions', () => {
        const Dimensions = {
          get: jest.fn().mockReturnValue({width: 100, height:100})
        }
        return Dimensions
      })
    }
  }