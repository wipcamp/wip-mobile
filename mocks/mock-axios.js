import axios from 'axios'
import mockAxios from 'axios-mock-adapter'

import env from '../config'

const mock = new mockAxios(axios)

export default mock