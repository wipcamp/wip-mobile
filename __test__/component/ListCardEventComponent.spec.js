/**
 * @jest-environment node
 */

import React from 'react'
import renderer from 'react-test-renderer'

import { timetable } from '../../mocks/data'

import ListCardEvent from '../../components/ListCardEventComponent'

it('rendering', () => {
    const tree = renderer
        .create(
            <ListCardEvent
                timetable={timetable}
                navigation={{}}
            />
        )
        .toJSON()
    expect(tree).toMatchSnapshot()
})