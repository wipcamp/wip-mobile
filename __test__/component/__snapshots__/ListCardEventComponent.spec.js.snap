// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering 1`] = `
<RCTScrollView
  DEPRECATED_sendUpdatedChildFrames={false}
  alwaysBounceHorizontal={false}
  alwaysBounceVertical={true}
  data={null}
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={null}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={undefined}
  onResponderTerminationRequest={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  onScrollShouldSetResponder={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  sendMomentumEvents={true}
  stickyHeaderIndices={Array []}
  style={
    Array [
      Object {
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "overflow": "scroll",
      },
      undefined,
    ]
  }
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <RCTScrollContentView
    collapsable={false}
    onLayout={[Function]}
    removeClippedSubviews={undefined}
    style={
      Array [
        false,
        undefined,
      ]
    }
  />
</RCTScrollView>
`;
